### ISSUE TITLE
Possible memory leak with ReferenceCountedOpenSslContext

### ISSUE THREAD

**Issue by dbakr** [jchambers/pushy#768]
Got this exception once in a while, still got no insight in which case it happens.
Will update you when know more.

Pushy Version 0.13.11

_Might be related to this:
`java.security.UnrecoverableKeyException: failed to decrypt safe contents entry: javax.crypto.BadPaddingException: Given final block not properly padded. Such issues can arise if a bad key is used during decryption.`_

```
20:09:59.622 [RxCachedThreadScheduler-212] ERROR io.netty.util.ResourceLeakDetector - LEAK: ReferenceCountedOpenSslContext.release() was not called before it's garbage-collected. See https://netty.io/wiki/reference-counted-objects.html for more information.
Recent access records: 
Created at:
	io.netty.handler.ssl.ReferenceCountedOpenSslContext.<init>(ReferenceCountedOpenSslContext.java:205)
	io.netty.handler.ssl.ReferenceCountedOpenSslContext.<init>(ReferenceCountedOpenSslContext.java:185)
	io.netty.handler.ssl.ReferenceCountedOpenSslClientContext.<init>(ReferenceCountedOpenSslClientContext.java:67)
	io.netty.handler.ssl.SslContext.newClientContextInternal(SslContext.java:833)
	io.netty.handler.ssl.SslContextBuilder.build(SslContextBuilder.java:576)
	com.eatthepath.pushy.apns.ApnsClientBuilder.build(ApnsClientBuilder.java:563)
	com.af.rta.uninstalls.apns.RxApnsSender.lambda$buildApnsClient$23(RxApnsSender.java:358)
	io.reactivex.rxjava3.internal.operators.single.SingleFromSupplier.subscribeActual(SingleFromSupplier.java:48)
	io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	io.reactivex.rxjava3.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:589)
	io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
	io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
	java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by jchambers** [comment]
Do you have a line (or full stack trace or other clue) for where that `UnrecoverableKeyException` is happening?

**Comment by jchambers** [comment]
@dbakr I hate to be a pest, but can you provide any more information? Understanding where your `UnrecoverableKeyException` is happening would be a big help. Thanks!

**Comment by dbakr** [comment]
I haven't received it for a week already... not sure which changes in my code lead to this, but I'm sorry I can't give you more info. Feel free to close the issue, if I ever start to encounter it again, I'll reopen with details.

**Comment by jchambers** [comment]
Thank you for following up! I did what I could with the information available in #774, and we'll hope that's enough for now.
